### PR TITLE
fix(sell) : fix issue #535

### PR DESCRIPTION
**Pull request by shmodarresi**
fix issue https://github.com/tzConnectBerlin/prediction-market-ui/issues/535

### CODE REVIEW

**Review by tlebon — DISMISSED** [review]
looks good

**Comment by shekhar-shubhendu** [comment]
@shmodarresi I can still see the issue. I'm unable to sell even though I have enough tokens and the market has liquidity.

https://prediction-market-git-feat-sell-bugs-tz-connect.vercel.app/market/7/is-this-a-new-test-market

**Comment by shmodarresi** [comment]
> @shmodarresi I can still see the issue. I'm unable to sell even though I have enough tokens and the market has liquidity.
> 
> https://prediction-market-git-feat-sell-bugs-tz-connect.vercel.app/market/7/is-this-a-new-test-market

So, let me check it again

**Comment by shekhar-shubhendu** [comment]
@shmodarresi @tlebon I've pushed a change that should fix this. Please test the app and see that you are able to sell

**Comment by shekhar-shubhendu** [comment]
the reasoning behind my fix:

image

here the slippage is applied to what we get back and not what we send in.

File: src/contracts/MarketCalculations.ts [hunk 1]
```diff
@@ -44,14 +44,14 @@ export const closePosition = (
   slippage: number,
 ): ClosePositionReturn => {
   const aToSwap = optimalSwap(aPool, bPool, aHoldings);
-  const aToSwapWithSlippage = aToSwap + (aToSwap * slippage) / 100;
-  const aLeft = aHoldings - aToSwapWithSlippage;
-  const bReceived = fixedInSwap(aPool, bPool, aToSwapWithSlippage);
+  const bReceived = fixedInSwap(aPool, bPool, aToSwap);
+  const bReceivedWithSlippage = bReceived - (bReceived * slippage) / 100;
```
Comment (shmodarresi) [inline src/contracts/MarketCalculations.ts:48]: @shekhar-shubhendu , the formula should still be `bReceived - (bReceived * slippage) / 100` or it should be changed to `bReceived + (bReceived * slippage) / 100`

Comment (shekhar-shubhendu) [reply src/contracts/MarketCalculations.ts:48]: `minus` since we are accepting `min_output`

**Comment by tlebon** [comment]
seems to be working fine now

**Review by tlebon — APPROVED** [review]

**Review by shmodarresi — COMMENTED** [review]
it works fine, thanks @shekhar-shubhendu

**Pull request merged**